src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -114,12 +112,13 @@ public boolean deleteById(Long id) {
         }
     }
 
-    private Book setBook(ResultSet resultSet) throws SQLException {
+    private Book getBookFromResultSet(ResultSet resultSet, Long id) throws SQLException {
         String title = resultSet.getString("title");
         BigDecimal price = resultSet.getObject("price", BigDecimal.class);
         Book book = new Book();
         book.setTitle(title);
         book.setPrice(price);
+        book.setId(id);
         return book;
     }
```

Comment:
do you really need to pass id param here? Isn't it possible to retrieve id from resultSet for both create and find operations?

Reply:
simple things the hardest)